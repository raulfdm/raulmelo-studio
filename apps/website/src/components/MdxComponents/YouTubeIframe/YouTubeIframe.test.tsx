import React from 'react';
import { render, screen } from '@testing-library/react';

import { YouTubeIframe } from '.';

describe('<YouTubeIframe />', () => {
  it('renders with videoId', () => {
    const videoId = '3-Pf8s2nP-g';

    render(<YouTubeIframe videoId={videoId} />);

    const container = screen.getByTestId('youtube-iframe') as HTMLIFrameElement;

    expect(container.src).toMatchInlineSnapshot(
      `"https://www.youtube.com/embed/3-Pf8s2nP-g"`,
    );
    expect(container).toMatchSnapshot();
  });

  it('throws an error if neither "src" nor "directUrl" has been passed', () => {
    expect(() => render(<YouTubeIframe />)).toThrowErrorMatchingInlineSnapshot(
      `"YouTubeIframe: \\"videoId\\" is required"`,
    );
  });
});
