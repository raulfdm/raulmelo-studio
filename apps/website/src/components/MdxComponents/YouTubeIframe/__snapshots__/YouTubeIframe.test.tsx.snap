// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YouTubeIframe /> renders with src 1`] = `
<iframe
  allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
  allowfullscreen=""
  class="absolute top-0 left-0 h-full w-full"
  data-testid="youtube-iframe"
  frameborder="0"
  height="315"
  src="https://www.youtube.com/embed/3-Pf8s2nP-g"
  title="YouTube Video"
  width="560"
/>
`;

exports[`<YouTubeIframe /> renders with videoId 2`] = `
<iframe
  allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
  allowfullscreen=""
  class="absolute top-0 left-0 h-full w-full"
  data-testid="youtube-iframe"
  frameborder="0"
  height="315"
  src="https://www.youtube.com/embed/3-Pf8s2nP-g"
  title="YouTube Video"
  width="560"
/>
`;
